Fantasy-Quest/Assets/Scripts/Utils/TransformMover.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Collections;
+using Rails;
+using Sirenix.OdinInspector;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Utils
+{
+    [AddComponentMenu("Scripts/Scripts/Utils/Utils.TransformMover")]
+    internal class TransformMover : MonoBehaviour
+    {
+        private enum Mode
+        {
+            Transform,
+            Vector3,
+            Point
+        }
+
+        [Required]
+        [SerializeField]
+        private Transform movingBody;
+
+        [EnumToggleButtons]
+        [SerializeField]
+        private Mode mode = Mode.Point;
+
+        [SerializeField]
+        [ShowIf(nameof(mode), Value = Mode.Transform)]
+        private Transform finalPosition;
+
+        [SerializeField]
+        [ShowIf(nameof(mode), Value = Mode.Vector3)]
+        private Vector3 finalPositionV3;
+
+        [SerializeField]
+        [ShowIf(nameof(mode), Value = Mode.Point)]
+        private Point finalPositionPoint;
+
+        [SerializeField]
+        private float time;
+
+        [SerializeField]
+        private bool useCurve = false;
+
+        [SerializeField]
+        [ShowIf(nameof(useCurve))]
+        private AnimationCurve curve;
+
+        public UnityEvent MoveFinished;
+
+        private void Update()
+        {
+            if (Input.GetKeyDown(KeyCode.R))
```

Comment:
Is it a correct button? Why new input system is not used?

Reply:
That was for test purpose, i'll remove this)